examples/cfd/swe_distributed_gnn/train.py

```diff
@@ -0,0 +1,640 @@
+            metrics=metrics,
+            num_examples=num_examples,
+            num_epochs=num_epochs,
+            loss_fn=loss_fn,
+            num_future=num_future,
+            enable_amp=enable_amp,
+        )
+
+        training_time = time.time() - start_time
+
+        if dist_manager.rank == 0:
+            os.makedirs(os.path.join(output_dir, run_suffix, "figures"), exist_ok=True)
+            os.makedirs(
+                os.path.join(output_dir, run_suffix, "output_data"), exist_ok=True
+            )
+            if save_checkpoint:
+                os.makedirs(
+                    os.path.join(output_dir, run_suffix, "checkpoints"), exist_ok=True
+                )
+                torch.save(
+                    model.state_dict(),
+                    os.path.join(output_dir, run_suffix, "checkpoints", "model.pt"),
+                )
+
+    # skip inference for dummy_data
+    if not dummy_data:
+        # set seed
+        torch.manual_seed(seed)
+        torch.cuda.manual_seed(seed)
+
+        with torch.inference_mode():
+            losses, fno_times, nwp_times = autoregressive_inference(
+                model,
+                dataset,
+                os.path.join(output_dir, run_suffix, "figures"),
+                dist_manager=dist_manager,
+                loss_fn=loss_fn,
+                nsteps=nsteps,
+                autoreg_steps=10,
+            )
+            metrics["loss_mean"] = np.mean(losses)
+            metrics["loss_std"] = np.std(losses)
+            metrics["fno_time_mean"] = np.mean(fno_times)
+            metrics["fno_time_std"] = np.std(fno_times)
+            metrics["nwp_time_mean"] = np.mean(nwp_times)
+            metrics["nwp_time_std"] = np.std(nwp_times)
+
+    max_mem = torch.tensor(
+        [
+            torch.cuda.max_memory_allocated() * 1.0 / (1024**3),
+            torch.cuda.max_memory_reserved() * 1.0 / (1024**3),
+        ],
+        dtype=torch.float32,
+        device=dist_manager.device,
+    ).view(1, 2)
+    if dist_manager.rank == 0:
+        gather_list = [
+            torch.empty_like(max_mem) for r in range(dist_manager.world_size)
+        ]
+    else:
+        gather_list = []
+
+    if DistributedManager().distributed:
+        torch.distributed.gather(max_mem, gather_list, dst=0)
+    else:
+        max_mem = gather_list[0]
+
+    if dist_manager.rank == 0:
+        if train:
+            max_mem = torch.vstack(gather_list)
+            metrics["training_time"] = training_time
+            metrics[f"max_memory_allocated_gib"] = max_mem[:, 0].tolist()
+            metrics[f"max_memory_reserved_gib"] = max_mem[:, 1].tolist()
+
+        with open(
+            os.path.join(output_dir, run_suffix, "output_data", "metrics.json"), "w"
+        ) as f:
+            json.dump(metrics, f)
+
+    if DistributedManager().distributed:
+        DistributedManager.cleanup()
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
```

Comment:
Need to use hydra configs instead of argparse. Please let me know if I can help with that.

Reply:
done